qiskit/synthesis/two_qubit/two_qubit_decompose.py

```diff
@@ -280,157 +279,29 @@ def __init__(self, rxx_equivalent_gate: Type[Gate]):
         Raises:
             QiskitError: If the gate is not locally equivalent to an :class:`.RXXGate`.
         """
-        atol = DEFAULT_ATOL
-
-        scales, test_angles, scale = [], [0.2, 0.3, np.pi / 2], None
-
-        for test_angle in test_angles:
-            # Check that gate takes a single angle parameter
-            try:
-                rxx_equivalent_gate(test_angle, label="foo")
-            except TypeError as _:
-                raise QiskitError("Equivalent gate needs to take exactly 1 angle parameter.") from _
-            decomp = TwoQubitWeylDecomposition(rxx_equivalent_gate(test_angle))
-
-            circ = QuantumCircuit(2)
-            circ.rxx(test_angle, 0, 1)
-            decomposer_rxx = TwoQubitWeylDecomposition(
-                Operator(circ).data,
-                fidelity=None,
-                _specialization=two_qubit_decompose.Specialization.ControlledEquiv,
-            )
-
-            circ = QuantumCircuit(2)
-            circ.append(rxx_equivalent_gate(test_angle), qargs=[0, 1])
-            decomposer_equiv = TwoQubitWeylDecomposition(
-                Operator(circ).data,
-                fidelity=None,
-                _specialization=two_qubit_decompose.Specialization.ControlledEquiv,
-            )
-
-            scale = decomposer_rxx.a / decomposer_equiv.a
-
-            if abs(decomp.a * 2 - test_angle / scale) > atol:
-                raise QiskitError(
-                    f"{rxx_equivalent_gate.__name__} is not equivalent to an RXXGate."
-                )
-
-            scales.append(scale)
-
-        # Check that all three tested angles give the same scale
-        if not np.allclose(scales, [scale] * len(test_angles)):
-            raise QiskitError(
-                f"Cannot initialize {self.__class__.__name__}: with gate {rxx_equivalent_gate}. "
-                "Inconsistent scaling parameters in checks."
-            )
-
-        self.scale = scales[0]
-
-        self.rxx_equivalent_gate = rxx_equivalent_gate
-
-    def __call__(self, unitary, *, atol=DEFAULT_ATOL) -> QuantumCircuit:
-        """Returns the Weyl decomposition in circuit form.
-
-        Note: atol ist passed to OneQubitEulerDecomposer.
-        """
-
-        # pylint: disable=attribute-defined-outside-init
-        self.decomposer = TwoQubitWeylDecomposition(unitary)
-
-        oneq_decompose = OneQubitEulerDecomposer("ZYZ")
-        c1l, c1r, c2l, c2r = (
-            oneq_decompose(k, atol=atol)
-            for k in (
-                self.decomposer.K1l,
-                self.decomposer.K1r,
-                self.decomposer.K2l,
-                self.decomposer.K2r,
-            )
+        self._inner_decomposition = two_qubit_decompose.TwoQubitControlledUDecomposer(
+            rxx_equivalent_gate._standard_gate
```

Comment:
This is potentially a breaking API change if we treat this class as public. I think we can probably get away with considering it private since as far as I can tell we've never documented it. If we wanted to support arbitrary gates it wouldn't be too hard this is why `TwoQubitGateSequence` has an `Option<StandardGate>` in it. We'd have to pass the other details to get the matrix from the parameter value to rust but then we can keep the arbitrary gate support and when we encounter `None` we use that.

That being said if we're ok (I'm not 100% convinced one way or the other) making this API change to the class because we consider this internal only, we should change this to do something like:

```suggestion
        if (gate := rxx_equivalent_gate.getattr("_standard_gate", None)) is not None:
            self._inner_decomposition = two_qubit_decompose.TwoQubitControlledUDecomposer(
                rxx_equivalent_gate._standard_gate
        else:
            raise QiskitError("Must be initialized with a standard gate object")
```

so we raise a meaningful error if this is instantiated with a non-standard gate.

Reply:
done in [3182838](https://github.com/Qiskit/qiskit/pull/13139/commits/31828388b0808cffd2bb7def11b70382fc50e595)
I had to change the if line to 
`if rxx_equivalent_gate._standard_gate is not None:`
otherwise the tests were failing.

Reply:
I restored the capability to use custom gate classes in: https://github.com/Qiskit/qiskit/pull/13139/commits/f46edbee26053d555d21283b9046b45207f863a5 It depends on calling back to python to instantiate a custom gate class but with this we don't have any concerns with any potential API change.